mautic/mautic
# cookie with samesite=none require secure

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes/no
| New feature?                           | yes/no
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9019




#### Description:

Cookies set from different domain should send SameSite=None attribute. New standard require also use it on https with param secure.

The PR prevent

> This Set-Cookie was blocked because it had the "SameSite=None" attribute but did not have the "Secure" attribute, which is required in order to use "SameSite=None".

From documentation https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite

> If SameSite=None is set, the cookie Secure attribute must also be set (or the cookie will be blocked).

The same we can expect also in chromium:

https://blog.chromium.org/2019/10/developers-get-ready-for-new.html



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. This works just on https 
3. You should check after PR all using cookies parts works properly - tracking code, form submission, focus...




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9743"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
